docs/reference/sql/security.asciidoc

```diff
@@ -24,13 +24,43 @@ PKI/X.509:: Use X.509 certificates to authenticate {es-sql} to {es}. For this, o
 [discrete]
 [[sql-security-permissions]]
 ==== Permissions (server-side)
-Lastly, one the server one need to add a few permissions to
-users so they can run SQL. To run SQL a user needs `read` and
+On the server, one needs to add a few permissions to
+users so they can run SQL. To run SQL, a user needs `read` and
 `indices:admin/get` permissions at minimum while some parts of 
-the API require `cluster:monitor/main`. 
+the API require `cluster:monitor/main`.
 
-The following example configures a role that can run SQL in JDBC querying the `test` and `bort`
-indices:
+You can add permissions with either an <<apiexample,API call>>
+or the configuration file <<fileexample,roles.yml>>. File-based role management
+is useful if you want to define a role that doesn't need to change. Also, you
+cannot use the role management APIs to view or edit a role defined in `roles.yml`.
+Using a role management API is the preferred method for defining roles.
+
+[[apiexample]]
+*Add permissions with the role management APIs*
+
+This example configures a role that can run SQL in JDBC querying the `test`
+index:
+
+[source,console]
+--------------------------------------------------
+POST /_security/role/cli_or_drivers_minimal
+{
+  "cluster": ["cluster:monitor/main"],
+  "indices": [
+    {
+      "names": ["test"],
+      "privileges": ["read", "indices:admin/get"]
+    }
+  ]
+}
+--------------------------------------------------
+// TEST[skip:forcomparisontofilemethod]
```

Comment:
I don't think we should skip this test

Reply:
Hi @abdonpijpelink 

I should be able to get this test running. Actually, I have a few questions about the approach.

When I put this API example in `create-roles.asciidoc`, it passes. However, when I run it in `security.asciidoc`
it fails with `error: java.lang.AssertionError: expected [2xx] status code but api [raw[method=POST path=_security/role/cli_or_drivers_minimal]] returned [400 Bad Request] [{"error":"no handler found for uri [/_security/role/cli_or_drivers_minimal?error_trace=true] and method [POST]"}]`

I'm going to make a few assumptions on how to solve this, but you can correct my understanding:

- Do I have to set up security in the `build.gradle` file? [the one directly under `elasticsearch/docs/`
directory? I saw another under `elasticsearch/x-pack/docs/` which I think has the security settings to enable the create-roles tests to pass. I tried mucking around with the x-pack settings in the `elasticsearch/docs/build.gradle` but did not make much progress].  
- Is there any internal documentation on how to set up the configuration for the tests ?
- What type of elasticsearch instance runs for these docs test cases? 

- If it's not a good idea to modify the `build.gradle` file, could I somehow "link" the API example from the `create-roles.asciidoc` page? Exactly like how the previous author does it with the `roles.yml` snippet below it (with `include-tagged`). For instance, the API example would be (hidden?) on the `create-roles.asciidoc` page and the test runs "there", but somehow it could be displayed on the `security.asciidoc` page by some Asciidoc link (do you know if something like that exists for API examples?)

Reply:
Great questions @anthonymcglone2022 

> Do I have to set up security in the build.gradle file?

No, that should not be necessary

> Is there any internal documentation on how to set up the configuration for the tests ?

All we have is here: https://github.com/elastic/elasticsearch/tree/main/docs#test-code-snippets

> What type of elasticsearch instance runs for these docs test cases?

Should be an instance with security enabled, as that is the default these days.

I don't understand why this test fails, and will have to look into it a bit more. The snippet is similar to [examples on other pages](https://www.elastic.co/guide/en/elasticsearch/reference/master/defining-roles.html#_example_8), and the test doesn't fail there. 

Maybe this test only fails locally somehow? One thing we can try is removing the `// TEST[skip:forcomparisontofilemethod]` line. I can then run a CI build and we can see what happens.

Reply:
@abdonpijpelink I was able to get the test running. I used `include::/path/to/file[tag=tag_id]` in the `security.asciidoc` file and used `// tag::tag_id[]` on the API snippet, which I placed in `create-roles.asciidoc` . It displays ok in `security.asciidoc` and the test runs. Do you think it's ok to push this up, or you would prefer me to continue the troubleshooting with the CI build (and so on)?

Reply:
Push it. We'll troubleshoot the test on our end. Thank you!